Replicator/c4Replicator.hh

```diff
@@ -136,7 +136,17 @@ struct C4Replicator : public RefCounted,
 
     C4ReplicatorStatus status() {
```

Comment:
I think that a much more robust thing to do would be to list, explicitly, the statuses that are returned, according to the doc, and to return them.
The default could be what kC4Stopping is now... or better yet, could throw an error, and kC4Stopping could be handled explicitly.

Reply:
That sounds reasonable, and new statuses would fail by being an incorrect but valid status instead of a potentially invalid one if they are not handled.  Sounds more futureproof.